pkg/security/middlewares/common.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package middlewares
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+type middleware struct {
+	FilterName string
+}
+
+// Name implements the web.Filter interface and returns the identifier of the filter
+func (m *middleware) Name() string {
+	return m.FilterName
+}
+
+// FilterMatchers implements the web.Filter interface and returns the conditions on which the filter should be executed
+func (m *middleware) FilterMatchers() []web.FilterMatcher {
+	return []web.FilterMatcher{}
+}
+
+// ErrUserNotFound error when authentication passed but no user found
+var ErrUserNotFound = errors.New("user identity must be provided when allowing authentication")
+
+// UnauthorizedHTTPError returns HTTPError 401 with description from given error
+func UnauthorizedHTTPError(err error) *util.HTTPError {
```

Comment:
why is this a function? Better make it a var as there's no use of the error parameter for now. Also making it a var makes it easier to type check the returned error (applies for the the next function also)